### PR TITLE
Technical/nijo 1003 contracts143

### PR DESCRIPTION
**Pull request by mingazhev**

### CODE REVIEW

**Review by hasmikharutyunyan — APPROVED** [review]

File: Api/Services/Accommodations/Availability/PriceProcessor.cs [hunk 1]
```diff
@@ -37,10 +38,10 @@ public PriceProcessor(IMarkupService markupService, ICurrencyConverterService cu
         {
             var markup = await _markupService.Get(agent, MarkupPolicyTarget.AccommodationAvailability);
             var responseWithMarkup = await priceProcessFunc(details, markup.Function);
-            var ceiledResponse = await priceProcessFunc(responseWithMarkup, (price, currency) =>
+            var ceiledResponse = await priceProcessFunc(responseWithMarkup, price =>
             {
-                var roundedPrice = MoneyRounder.Ceil(price, currency);
-                return new ValueTask<(decimal, Currencies)>((roundedPrice, currency));
+                var ceiledAmount = MoneyRounder.Ceil(price.Amount, price.Currency);
```
Comment (kirillta) [inline Api/Services/Accommodations/Availability/PriceProcessor.cs:43]: The `MoneyRounder` can work with a `MoneyAmount` directly

File: Api/Services/CurrencyConversion/CurrencyConverterService.cs [hunk 2]
```diff
@@ -33,12 +34,12 @@ public CurrencyConverterService(ICurrencyRateService rateService)
             if (isFailure)
                 return Result.Failure<TData>(error);
 
-            var convertedDetails = await changePricesFunc(data, (price, currency) =>
+            var convertedDetails = await changePricesFunc(data, price =>
             {
-                var newPrice = price * rate * (1 + ConversionBuffer);
-                var ceiledPrice = MoneyRounder.Ceil(newPrice, TargetCurrency);
+                var newAmount = price.Amount * rate * (1 + ConversionBuffer);
```
Comment (kirillta) [inline Api/Services/CurrencyConversion/CurrencyConverterService.cs:39]: In an ideal world the `rate` and the `ConversionBuffer` must be type of _MoneyAmount_

Comment (mingazhev) [reply Api/Services/CurrencyConversion/CurrencyConverterService.cs:39]: They are not absolute values.
And there is another special task for make conversion buffers and other things more consistent

File: Api/Services/Accommodations/Availability/PriceProcessor.cs (hunk 1 above)
Comment (mingazhev) [reply Api/Services/Accommodations/Availability/PriceProcessor.cs]: Fixed

**Review by kirillta — APPROVED** [review]

**Review by Raibo — APPROVED** [review]

**Pull request merged**